clients/gtest/rocblas_gtest_main.cpp

```diff
@@ -221,31 +247,50 @@ void catch_signals_and_exceptions_as_failures(const std::function<void()>& test)
     // Save this thread's id, to detect signals in different threads
     rocblas_test_sighandler_tid = pthread_self();
 
-    // Set up the return point
-    int sig = sigsetjmp(rocblas_sigjmp_buf, true);
+    // Catch std::terminate
+    // old_terminate must be volatile to survive across sigsetjmp
+    volatile auto old_terminate = std::set_terminate(terminate_handler);
 
-    // If this is a return, indicate the signal received
-    if(sig)
+    // Set up the return point, and handle siglongjmp returns back to here
+    switch(sigsetjmp(rocblas_sigjmp_buf, true))
```

Comment:
We cannot store the return value of `sigsetjmp` into a variable and be 100% standards compliant -- its return value can only be used as the controlling expression of a control statement, or discarded.